### PR TITLE
Linting package on CI

### LINKED ISSUE #5: Add CI workflow
We should have CI to verify our code, such as package lint, check doc and byte compile. It would be great if we can do with Earthly by running emacs inside the container and use setup Earthly in GitHub workflow to install Earthly inside it.

### PR DESCRIPTION
**Pull request by wingyplus**
This changes introduce Eldev to uses for lingting the package and add
Earthfile to run Eldev inside the container. And add GitHub workflow
to run linting against Earthly.

The CI workflow run on 2 Emacs version, 26 which's minimum support and
27 which's latest stable version.

You can see an example [here](https://github.com/wingyplus/earthly-mode-1/actions/runs/1179059981).

Closes #5

### CODE REVIEW

**Comment by wingyplus** [comment]
@vladaionescu request reviewing. :)

File: .github/workflows/ci.yaml
Comment (wingyplus) [inline .github/workflows/ci.yaml:21]: Do we need to config the registry mirror like earthly repository does?

Ref: https://github.com/earthly/earthly/blob/main/.github/workflows/ci.yml#L23-L31

**Comment by vladaionescu** [comment]
Hi @wingyplus - sorry for the delay. Will take a look at this next week most likely since I need to adjust some CI-related permissions first.

(Separately, I've given you write access to this repo so you can manage/review any incoming PRs yourself directly if needed. We should aim for at least 1 PR review, and ideally, any new contributions should be manually tested by the reviewer as well, to make sure the package is not broken. Thanks for making Earthly better!!)

**Comment by wingyplus** [comment]
> Hi @wingyplus - sorry for the delay. Will take a look at this next week most likely since I need to adjust some CI-related permissions first.
> 
> (Separately, I've given you write access to this repo so you can manage/review any incoming PRs yourself directly if needed. We should aim for at least 1 PR review, and ideally, any new contributions should be manually tested by the reviewer as well, to make sure the package is not broken. Thanks for making Earthly better!!)

@vladaionescu Thank you for your invitation.  ^^

**Comment by wingyplus** [comment]
And this PR is just nice to have at the moment, review it when you have a free time. :)

File: Earthfile
Comment (vladaionescu) [inline Earthfile:1]: It's possible to do the matrix testing within earthly itself. For example you could define a target like this:

```
lint-all:
  BUILD +lint --EMACS_VERSION=26 --EMACS_VERSION=27
```

**Review by vladaionescu — APPROVED** [review]
This looks pretty good - just a mild preference for the matrix to be within Earthly

**Comment by wingyplus** [comment]
Sorry for the delay. I would like to merge this and will refactor after that since I accidentally remove the fork repo both my github account and local machine. 🙇

**Pull request merged**
